torchdata/datapipes/iter/util/prefetcher.py

```diff
@@ -101,6 +101,8 @@ def __iter__(self):
                 if self.thread is not None:
                     self.thread.join()
                     self.thread = None
+                for sample in prefetch_data.prefetch_buffer:
+                    yield sample
```

Comment:
I think it's not the right place. Let's move it before `finally` but after `while` loop